sim/midas/src/main/cc/core/widget_registry.h

```diff
@@ -56,10 +56,23 @@ class widget_registry_t final {
    */
   template <typename T>
   T &get_widget() {
+    auto *widget = get_widget_opt<T>();
+    assert(widget && "cannot find widget");
+    return *widget;
+  }
+
+  /**
+   * Return a widget of a particular kind, if it has a single instance.
+   *
+   * @tparam T Type of the widget to fetch
+   * @return Pointer to the widget instance.
+   */
+  template <typename T>
+  T *get_widget_opt() {
```

Comment:
Why use a pointer instead of `std::optional`? (for my own elucidation)

Reply:
We want a non-owning reference here and a pointer is the way to do it (I don't want a shared_pointer since the registry is the owner of the objects). `std::optional` should be used with value types.